python/ray/runtime_env.py

```diff
@@ -26,3 +44,497 @@ def get_current_runtime_env():
+
+
+def _parse_proto_conda_runtime_env(
+    runtime_env: ProtoRuntimeEnv, runtime_env_dict: dict
+):
+    """Parse conda runtime env protobuf to runtime env dict."""
+    if runtime_env.python_runtime_env.HasField("conda_runtime_env"):
+        conda_runtime_env = runtime_env.python_runtime_env.conda_runtime_env
+        if conda_runtime_env.HasField("conda_env_name"):
+            runtime_env_dict["conda"] = conda_runtime_env.conda_env_name
+        else:
+            runtime_env_dict["conda"] = json.loads(conda_runtime_env.config)
+
+
+def _build_proto_container_runtime_env(
+    runtime_env_dict: dict, runtime_env: ProtoRuntimeEnv
+):
+    """Construct container runtime env protobuf from runtime env dict."""
+    if runtime_env_dict.get("container"):
+        container = runtime_env_dict["container"]
+        runtime_env.python_runtime_env.container_runtime_env.image = container.get(
+            "image", ""
+        )
+        runtime_env.python_runtime_env.container_runtime_env.worker_path = (
+            container.get("worker_path", "")
+        )
+        runtime_env.python_runtime_env.container_runtime_env.run_options.extend(
+            container.get("run_options", [])
+        )
+
+
+def _parse_proto_container_runtime_env(
+    runtime_env: ProtoRuntimeEnv, runtime_env_dict: dict
+):
+    """Parse container runtime env protobuf to runtime env dict."""
+    if runtime_env.python_runtime_env.HasField("container_runtime_env"):
+        runtime_env_dict["container"][
+            "image"
+        ] = runtime_env.python_runtime_env.container_runtime_env.image
+        runtime_env_dict["container"][
+            "worker_path"
+        ] = runtime_env.python_runtime_env.container_runtime_env.worker_path
+        runtime_env_dict["container"]["run_options"] = list(
+            runtime_env.python_runtime_env.container_runtime_env.run_options
+        )
+
+
+def _build_proto_plugin_runtime_env(
+    runtime_env_dict: dict, runtime_env: ProtoRuntimeEnv
+):
+    """Construct plugin runtime env protobuf from runtime env dict."""
+    if runtime_env_dict.get("plugins"):
+        for class_path, plugin_field in runtime_env_dict["plugins"].items():
+            plugin = runtime_env.python_runtime_env.plugin_runtime_env.plugins.add()
+            plugin.class_path = class_path
+            plugin.config = json.dumps(plugin_field, sort_keys=True)
+
+
+def _parse_proto_plugin_runtime_env(
+    runtime_env: ProtoRuntimeEnv, runtime_env_dict: dict
+):
+    """Parse plugin runtime env protobuf to runtime env dict."""
+    if runtime_env.python_runtime_env.HasField("plugin_runtime_env"):
+        plugins: Dict[str, Any] = {}
+        for plugin in runtime_env.python_runtime_env.plugin_runtime_env.plugins:
+            plugins[plugin.class_path] = dict(json.loads(plugin.config))
+        if plugins:
+            runtime_env_dict["plugins"] = plugins
+
+
+@PublicAPI
+class RuntimeEnv(dict):
+    """This class use for define Runtime Environment.
```

Comment:
<img width="232" alt="Screen Shot 2022-02-23 at 11 08 40 AM" src="https://user-images.githubusercontent.com/5459654/155390088-59920201-4ec2-4a35-aa51-738009014498.png">
Any ideas how to fix this "Parameters" formatting issue?  If it's not straightforward we can leave it as a followup.  It might be some Sphinx issue that requires a creative workaround.  (I have a feeling it's because the code samples that appear in this block are too wide.)